db/migrate/20220207131452_create_groups.rb

```diff
@@ -0,0 +1,11 @@
+class CreateGroups < ActiveRecord::Migration[7.0]
+  def change
+    create_table :groups do |t|
+      t.string :name
+      t.string :icon
+      t.references :user, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```

Comment:
- [ ] Kindly note that the reference column should be `t.references :author, foreign_key: { to_table: 'users' }`.  This will create `author_id` and a corresponding index.